seed_swap.py

```diff
@@ -1,26 +1,35 @@
 from pipeline_reconstructor import PipelineReconstructor
 import argparse
 import json
-from query import find_pipelines
+from query import find_pipelines, get_pipeline
 from experimenter.execute_pipeline import execute_pipeline_on_problem
 from experimenter.problem import ProblemReference
 from d3m.metadata.pipeline import Pipeline
 import logging
 import logging.config as log_config
+from random import seed
+from random import randint
 
 
 logger = logging.getLogger(__name__)
 
 def main(**cli_args):
-    #create the function in the query file that returns the seeds that have already been run on and the pipeline to run on
-    seed_runs, pipeline = get_pipeline_seed(pipeline_id)
+    #get the relevant info from the query                               
+    pipeline, datasets_used, used_random_seeds = get_pipeline("f5478b6c-daad-4693-93c3-f073bae78ddf")
+    #now check if there are no used random seeds
     pipeline = json.dumps(pipeline, indent=4)
-    #now run the pipelines with the new seeds
-    for seed in cli_args['seeds']:
-        if (seed is in seed_runs):
-            logger.info("Seed {} already has a run on the pipeline".format(seed))
+    #now run the pipelines with new generated seeds
+    num_run = 0
+    #start the random seed
+    seed(cli_args['random_seed'])
+    #run pipelines until the correct number has been run
+    while (num_run <= cli_args['num_new_seeds']):
+        seed_num = randint(1,10000)
+        if (seed_num in used_random_seeds):
+            num_run = num_run
         else:
-            run_pipeline_seed(pipeline, seed, cli_args)
+            num_run += 1
+            run_pipeline_seed(pipeline, seed_num, **cli_args)
 
 def run_pipeline_seed(pipeline, seed, **cli_args):
     pipeline = Pipeline.from_json(string_or_file=pipeline)
```

Comment:
What if you called `from_json_structure()` instead? Then you wouldn't need to even use `json.dumps()`. Is there a reason to convert the pipeline to a string?
```suggestion
    pipeline = Pipeline.from_json_structure(pipeline)
```

Reply:
Yes, when working with pipelines, it should either be a file path or a json-like object.